tree/ntuple/v7/src/RPageStorageFile.cxx

```diff
@@ -380,11 +380,10 @@ ROOT::Experimental::Detail::RPage ROOT::Experimental::Detail::RPageSourceFile::P
       if (!cachedPage.IsNull())
          return cachedPage;
 
+      pageBuffer = new unsigned char[bytesPacked];
```

Comment:
Could you add a comment on why moving the allocation is solving the memory leak?  (The only obviously deletion is within the "seemingly" unrelated `!element->IsMappable()` and even-though I suspect some calls might lead to adoption, I don't find it/them trivially).

Reply:
The problem is in line 381: Before the patch, we could return early having the buffer already allocated (comment added).

Reply:
In addition to this move (not allocating it when not used), would it make sense to update `pageBuffer` to be a unique_ptr?

Reply:
@pcanal Sorry for leaving this dormant for so long!  It became more attention as the memory leak fixed was also found by clang-tidy.

Changed to a unique pointer, it does help clarifying ownership.